### PR TITLE
feat: add TypeScript definition and tests

### LINKED ISSUE #1: Write tests

### PR DESCRIPTION
**Pull request by arthurdenner**
Hi! 👋 

This PR adds TS definition, tests and also fixes the linter config.
Since `react-scripts` already installs ESLint and have its own config, we just need to extend it.
I've also added the Prettier integration to the project.

Plus, `wordsPerMinute` was added to the list of dependencies as suggested by ESLint itself.
As an example of a valid use case for it, imagine the user could pick their preferred speed of reading or something similar.

Closes #1.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -49,15 +51,10 @@
...
-    "eslint-plugin-promise": "^4.0.0",
-    "eslint-plugin-react": "^7.10.0",
-    "eslint-plugin-standard": "^3.1.0",
+    "eslint-config-prettier": "^6.11.0",
+    "eslint-plugin-prettier": "^3.1.4",
     "gh-pages": "^2.0.1",
+    "prettier": "^2.0.5",
     "react": "^16.9.0",
     "react-dom": "^16.13.1",
     "react-scripts": "^3.0.1",
```
Comment (gabrielferreiraa) [inline package.json:60]: @arthurdenner this three lines its not in your changes, but.. `react`, `react-dom` and `react-script` makes sense in `devDependencies`? `react` already be in `peerDependencies`.

**Review by gabrielferreiraa — COMMENTED** [review]
Amazing PR 🎉 
You were faster than me, I'd start this feature to add Typescript soon hahahahah Congrats ❤️

File: package.json (hunk 1 above)
Comment (arthurdenner) [reply package.json:60]: Yes, they should be declared as `devDependencies`.

**Comment by arthurdenner** [comment]
@diogomoretti, I hope you don't mind but I had written some tests yesterday, so I just polished them and pushed on this PR too.

The diff numbers are "big", but that's because `yarn.lock` was updated and I ran Prettier on the files 😶

**Comment by arthurdenner** [comment]
@diogomoretti ping.

**Comment by arthurdenner** [comment]
@diogomoretti, are you interested in this PR? Otherwise we can close it.

**Pull request closed without merging**